cardinal/message/message.go

```diff
@@ -228,6 +232,21 @@ func (t *MessageType[In, Out]) GetInFieldInformation() map[string]any {
 	return types.GetFieldInformation(reflect.TypeOf(new(In)).Elem())
 }
 
+func emitReceiptEvent(wCtx engine.Context, txHash types.TxHash, result any, errs []error) {
+	if err := wCtx.EmitEvent(
+		map[string]any{
+			"txHash": txHash,
+			"result": result,
+			"errs":   errs,
+		},
```

Comment:
This is a draft so just leaving a comment here. For this we should expose EmitJSONEvent to engine.Context to avoid this extra step of map conversion. 

Ideally this exposure should be "private". When users themselves want to launch an event they can use EmitEvent for type safety across nakama as well. I'm a little undecided about this as EmitJSONEvent eliminates an extra conversion to map step, but it sacrifices static safety as a result so not sure which one (or both) to make public.